lib/puppet/provider/ec2_instance/v2.rb

```diff
@@ -328,11 +350,44 @@ def stop
     @property_hash[:ensure] = :stopped
   end
 
+  def security_groups=(value)
+    @property_flush[:security_groups] = value
+  end
+
   def destroy
     Puppet.info("Deleting instance #{name} in region #{target_region}")
     ec2 = ec2_client(target_region)
     ec2.terminate_instances(instance_ids: [instance_id])
     ec2.wait_until(:instance_terminated, instance_ids: [instance_id])
     @property_hash[:ensure] = :absent
   end
+
+  def flush
+    if @property_hash[:ensure] != :absent
+      Puppet.debug("Flushing EC2 instance #{@property_hash[:name]}")
+      ec2 = ec2_client(resource[:region])
+
+      # Check if we need to modify the security groups for an interface
+      if @property_flush.keys.include? :security_groups
+        if @property_hash[:interfaces].size != 1
+          Puppet.warning('Handling security group changes on instances with
+                         multiple interfaces is not implemented')
+        else
+          interface_id = @property_hash[:interfaces].keys[0]
+          group_ids = ec2.describe_security_groups({
+            filters: [
+              {name: 'group-name', values: @property_flush[:security_groups]},
```

Comment:
There are other places in this module where various AWS resources can be referenced by ID. Do want to support referencing security groups by ID here?

(I'm thinking no.)